dirsrvtests/tests/suites/basic/basic_test.py

```diff
@@ -607,6 +625,85 @@ def test_basic_searches(topology_st, import_example_ldif):
     log.info('test_basic_searches: PASSED')
 
 
+@pytest.mark.parametrize('limit,resp',
+                         ((('200'), 'PASS'),
+                         (('50'), ldap.ADMINLIMIT_EXCEEDED)))
+def test_basic_search_lookthroughlimit(topology_st, limit, resp, import_example_ldif):
+    """Tests basic search with lookthroughlimit set high and low.
+
+    :id: b5119970-6c9f-41b7-9649-de9233226fec
+
+    :setup: Standalone instance, add example.ldif to the database (151 matches using (uid=*) filter)
+
+    :steps:
+         1. Import ldif user file.
+         2. Change lookthroughlimit to 200.
+         3. Create low priv user.
+         3. Bind to server as low priv user
+         4. Run a search with "high" lookthroughlimit.
+         5. Change lookthroughlimit to 50.
+         6. Run a search with "low" lookthroughlimit.
+         7. Bind to server as DM
+         8. Delete user from DB.
+         9. Reset lookthroughlimit to original.
+
+    :expectedresults:
+         1. First search should complete with no error.
+         2. Second search should return ldap.ADMINLIMIT_EXCEEDED error.
+    """
+
+    log.info('Running test_basic_search_lookthroughlimit...')
+
+    search_filter = r'(uid=*)'
+
+    ltl_orig = change_conf_attr(topology_st, 'cn=config,cn=ldbm database,cn=plugins,cn=config', 'nsslapd-lookthroughlimit', limit)
+
+    try:
+        topology_st.standalone.add_s(Entry((USER1_DN,
+                                            {'objectclass': "top extensibleObject".split(),
```

Comment:
Sorry I didn't see this earlier.  We are not using any of the "_s" functions from python-ldap in our test cases.  Everything should be a DSLdapObject.  So no "add_s()", "search_s", "modify_s", etc.  You should be able to use UserAccount/UserAccounts for most of this.   If you need help, Simon or myself can help you converting this over to DSLdapObject.

Reply:
Done